edi_bank_statement_import/tests/edi_bank_statement_import_test.py

```diff
@@ -0,0 +1,161 @@
+
+import base64
+import difflib
+import pprint
+import tempfile
+from datetime import date
+
+from odoo.modules.module import get_module_resource
+from odoo.tests.common import TransactionCase
+
+
+class TestParser(TransactionCase):
+    """Tests for the camt parser itself."""
+
+    def setUp(self):
+        super(TestParser, self).setUp()
+        self.parser = self.env["account.statement.import.camt.parser"]
+
+    def _do_parse_test(self, inputfile, goldenfile):
+        testfile = get_module_resource(
+            "account_statement_import_camt", "test_files", inputfile
+        )
+        with open(testfile, "rb") as data:
+            res = self.parser.parse(data.read())
+            with tempfile.NamedTemporaryFile(mode="w+", suffix=".pydata") as temp:
+                pprint.pprint(res, temp, width=160)
+                goldenfile_res = get_module_resource(
+                    "account_statement_import_camt", "test_files", goldenfile
+                )
+                with open(goldenfile_res, "r") as golden:
+                    temp.seek(0)
+                    diff = list(
+                        difflib.unified_diff(
+                            golden.readlines(), temp.readlines(), golden.name, temp.name
+                        )
+                    )
+                    if len(diff) > 2:
+                        self.fail(
+                            "actual output doesn't match expected "
+                            + "output:\n%s" % "".join(diff)
+                        )
+
+    def test_parse(self):
+        self._do_parse_test("test-camt053", "golden-camt053.pydata")
+
+    def test_parse_txdtls(self):
+        self._do_parse_test("test-camt053-txdtls", "golden-camt053-txdtls.pydata")
+
+    def test_parse_no_ntry(self):
+        self._do_parse_test("test-camt053-no-ntry", "golden-camt053-no-ntry.pydata")
+
+
+class TestImport(TransactionCase):
+    """Run test to import camt import."""
+
+    transactions = [
+        {
+            "account_number": "NL46ABNA0499998748",
+            "amount": -754.25,
+            "date": date(year=2014, month=1, day=5),
+            "ref": "435005714488-ABNO33052620",
+        },
+        {
+            "remote_account": "NL46ABNA0499998748",
+            "transferred_amount": -564.05,
+            "value_date": date(year=2014, month=1, day=5),
+            "ref": "TESTBANK/NL/20141229/01206408",
+        },
+        {
+            "remote_account": "NL46ABNA0499998748",
+            "transferred_amount": -100.0,
+            "value_date": date(year=2014, month=1, day=5),
+            "ref": "TESTBANK/NL/20141229/01206407",
+        },
+        {
+            "remote_account": "NL69ABNA0522123643",
+            "transferred_amount": 1405.31,
+            "value_date": date(year=2014, month=1, day=5),
+            "ref": "115",
+        },
+    ]
+
+    def setUp(self):
+        super(TestImport, self).setUp()
+        bank = self.env["res.partner.bank"].create(
```

Comment:
this code runs for every test. Please use `SavepointCase` + `setUpClass` and if you don't rely on tracking also use this line:
```
cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```
You'll see this approach in edi_oca too.

Reply:
removed the tests